### DISCUSSION TITLE
To improve the encryption and decryption algorithm strength,  change the salt value length to 16 bits.

### DISCUSSION THREAD

**Discussion by SperChung-11** [grafana/grafana#42038] [Platform]
hi, we have check the source code of Grafana and found that the salt used for encrypting the Grafana account is a 10-digit character string. Is the length of salt not enough for security? we think that the salt value could be changed to 16 characters to improve the encryption and decryption algorithm strength. somebody help ?

![image](https://user-images.githubusercontent.com/74529952/105851614-4baf5800-601e-11eb-9e09-ffd857ede216.png)

**Comment by sakjur** [comment]
> the salt used for encrypting the Grafana account is a 10-digit character string. Is the length of salt not enough for security? we think that the salt value could be changed to 16 characters to improve the encryption and decryption algorithm strength.

The characters is taken from an alpanumeric alphabet consisting of 62 characters, so each random character from that set adds `log2(62) ~= 5.95` bits. This means that we currently should have ~59.5 bits of entropy, and ~95.2 if we increased the number of characters from 10->16.

59.5 bits of entropy is _a very big number_, and it is highly unlikely that the same number will be generated twice with such a large set of salts (and even less likely for the same password). It doesn't hurt adding a few extra characters to the salt, but I don't think it'll have any measurable impact on security, as what we need here is something unique and 60 bits _should_ give plenty of room for uniqueness here.

**Comment by aknuds1** [comment]
So, based on @sakjur's comment, we need to figure out if this request actually makes sense.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
